DietPi-Software | Allow uninstall order by switching to method of install loop

MichaIng opened the pull request:
**WIP: Needs testing and review, it's more a method presentation than a pull "request"!**
+ Method adds new `aSOFTWARE_INSTALL_STATE[$1]=-1` to mark for uninstall. Then use `Uninstall_Software` one time to handle all marked titles in one loop. Allows uninstall order by code order, same as with install/configuration 👍.
+ Adds a lot of lines 👎.
+ Currently uninstall logging is disabled, does not work the old way. On the other hand we also do not have an install log (besides G_AGI calls) 👎.
+ Possibly allows to merge certain loops of install and uninstall, eg writing install state array to file 👍.

🈯️ Tested with installation + uninstallation of Nextcloud + pre-reqs, via GUI and cmd. Syntax seems okay now, method works.

Commit:
v6.3

+ DietPi-Software | Minor quoting and coding rearrangements
+ DietPi-Software | p7zip-full is DietPi core dependency, thus does not need to be installed and should not be removed on desktop uninstall

Commit:
v6.3

+ DietPi-Software | Coding: Prefer "grep <pattern> <filename>" over "cat <filename> | grep <pattern"

Commit:
v6.3

+ DietPi-Software | Allow uninstall order by switching to method of install loop

Commit:
Merge branch 'testing' into dietpi_software_uninstall_order

Commit:
v6.3

+ tiny

Commit:
v6.3

+ Parenthesis fix

Commit:
v6.3

+ Fix MariaDB uninstall: start service before mysqldump and avoid G_RUN_CMD when redirecting output to file

Commit:
Merge branch 'testing' into dietpi_software_uninstall_order

Commit:
v6.3

+ DietPi-Software | Uninstall_NonSelected_Choices: Mark software only if installed and add "Uninstall_Software"

Fourdee commented:
@MichaIng 

Impressive work, looks good 👍 

> 

Currently, only the 1st run of ```dietpi-software``` is logged (via ```login```):
```
/DietPi/dietpi/dietpi-software | tee "$FP_DIETPI_FIRSTRUNSETUP_LOG"
```

For software installs, we could try something like?:
```
Run_Installations | tee install.log
```

Uninstalls:
```
Uninstall_Software | tee uninstall.log
```

Fourdee commented:
REF (merge conflicts) https://github.com/Fourdee/DietPi/commit/7c237c30162b8e6380ac048afa176188d14f070f

Commit:
Merge branch 'testing' into dietpi_software_uninstall_order

Commit:
v6.5

+ DietPi-Software | Minor cleanup, aSOFTWARE_INSTALL_STATE[$i]=-1 marks UNINSTALL_FILE obsolete

Commit:
Merge branch 'testing' into dietpi_software_uninstall_order

Fourdee commented:
@MichaIng 

Legend, thanking you 👍

Fourdee commented:
@MichaIng 

Confirmed with testing, seems to work a treat 👍
